airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -146,6 +152,8 @@ def read(
                     logger.info(f"Finished syncing {configured_stream.stream.name}")
                     logger.info(timer.report())
 
+        if self.continue_sync_on_stream_failure and len(stream_name_to_exception) > 0:
+            raise AirbyteTracedException(message=self._generate_failed_streams_error_message(stream_name_to_exception))
```

Comment:
Testing this on the platform. This will yield a partial success message, but it will retry the sync. Successful streams that are incremental shouldn't resync previously committed records based on the emitted state.
<img width="1198" alt="Screenshot 2023-12-07 at 12 49 39 AM" src="https://github.com/airbytehq/airbyte/assets/51336873/10636863-c7ad-4130-9ab8-3f9e0ed5cbb8">